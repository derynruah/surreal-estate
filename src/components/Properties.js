import React, { useState, useEffect } from "react";
import axios from "axios";
import Alert from "./Alert";
import PropertyCard from "./PropertyCard";
import "../styles/properties.css";

const Properties = () => {
  const [properties, setProperties] = useState([]);
  const [alert, setAlert] = useState({ message: "" });

  useEffect(() => {
    axios
      .get("https://localhost:3000/v1/api/PropertyListing")
      .then(({ data }) => {
        setProperties(data);
        setAlert({ message: "" });
      })
      .catch(() =>
        setAlert({
          message: "Server error, please try again later.",
        })
      );
  }, []);

  return (
    <div className="properties-header">
      <h1>Properties</h1>
      <Alert message={alert.message} success={alert.isSuccess} />
      <div className="properties">
        {properties.map((property) => (
          <div key={property._id} className="item">
            <PropertyCard {...property} />
          </div>
        ))}
      </div>
    </div>
  );
};

export default Properties;
