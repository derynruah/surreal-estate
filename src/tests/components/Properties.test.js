import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { render } from "@testing-library/react";
import Properties from "../../components/Properties";

const RouterRender = () => {
  return (
    <Router>
      <Properties />
    </Router>
  );
};

describe("Properties", () => {
  test("renders correctly", () => {
    const { asFragment } = render(<RouterRender />);

    expect(asFragment()).toMatchSnapshot();
  });
});
