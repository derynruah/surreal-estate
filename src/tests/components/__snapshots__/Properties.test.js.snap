// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Properties renders correctly 1`] = `
<DocumentFragment>
  <div
    class="properties-header"
  >
    <h1>
      Properties
    </h1>
    <div
      class="properties"
    />
  </div>
</DocumentFragment>
`;
